addons/hitreactions/XEH_postInit.sqf

```diff
@@ -0,0 +1,68 @@
+#include "script_component.hpp"
+
+GVAR(armComponents) = [
+    "leftshoulder",
+    "rightshoulder",
+    "lefthand",
+    "leftforearm",
+    "leftarmroll",
+    "rightforearm",
+    "rightarmroll",
+    "righthand",
+    "rightarm"
+];
+
+GVAR(safePickupDistance) = 8;
+
+["ace_firedNonPlayer", {
+    params ["", "", "", "", "", "", "_projectile"];
+    if (GVAR(weaponDropChanceGunHit) + GVAR(weaponDropChanceArmHit) == 0) exitWith {};
+    private _roll = random 1;
+    private _willDropGun = _roll < GVAR(weaponDropChanceGunHit);
+    private _willDropArm = _roll < GVAR(weaponDropChanceArmHit);
```

Comment:
I wouldn't do variable allocation and just throw these into the condition block directly

Reply:
More readable imo

Reply:
Performance is a concern if this will be used on fired event.

Reply:
Done